janusgraph-core/src/main/java/org/janusgraph/graphdb/relations/CacheVertexProperty.java

```diff
@@ -133,7 +133,7 @@ public byte getLifeCycle() {
 
     @Override
     public void remove() {
-        if (!tx().isRemovedRelation(super.longId())) {
+        if (!tx().isRemovedRelation(super.longId()) && !tx().getConfiguration().isReadOnly()) {
```

Comment:
This was added to resolve test failures in the below tests and I think throwing the exception would also cause failures. [TraversalVertexProgram](https://github.com/apache/tinkerpop/blob/3.2.4/gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/process/computer/traversal/TraversalVertexProgram.java#L294-L296) has a convenience property removal that causes this issue. Ideally if possible logic could be added there to determine if transaction is readOnly.

`org.apache.tinkerpop.gremlin.process.computer.GraphComputerTest#shouldFailWithImproperTraverserRequirements`
`org.apache.tinkerpop.gremlin.process.computer.GraphComputerTest#shouldSucceedWithProperTraverserRequirements`